app/controllers/items_controller.rb

```diff
@@ -18,17 +19,30 @@ def new
   def create
     item = Item.new(item_params)
     item.seller_id = current_user.id
-    @presenter = NewItemPresenter.new(item)
-    if @presenter.item.save
+    if item.valid?
+      item.save
+      if item.auction_end_time && item.starting_bid_price
+        AuctionWorker.perform_in(auction_duration(item).seconds, item.id)
+      end
       redirect_to items_path
     else
+      @presenter = NewItemPresenter.new(item)
       render :new, status: :unprocessable_entity
     end
   end
 
+  def show
+    item = Item.find(params[:id])
+    @presenter = ItemPresenter.new(item)
+  end
+
   private
 
   def item_params
-    params.require(:item).permit(:name, :description, :price, :image)
+    params.require(:item).permit(:name, :description, :buy_it_now_price, :starting_bid_price, :auction_end_time, :image)
+  end
+
+  def auction_duration(item)
```

Comment:
Doesn't this method return seconds already? Does line 25 need to call `.seconds` on the return value?